doc/topics/the-event-loop-timers-and-nexttick

```diff
@@ -0,0 +1,174 @@
+# Overview of the Event Loop, Timers, and `process.nextTick()`
+
+The Following diagram shows a simplified overview of the event loop's order of operations.
+
+       ┌───────────────────────┐
+    ┌─>│        timers         │
+    │  └──────────┬────────────┘
+    │  ┌──────────┴────────────┐
+    │  │   pending callbacks   │
+    │  └──────────┬────────────┘      ┌───────────────┐
+    │  ┌──────────┴────────────┐      │   incoming:   │
+    │  │         poll          │<─────│  connections, │
+    │  └──────────┬────────────┘      │   data, etc.  │
+    │  ┌──────────┴────────────┐      └───────────────┘
+    └──│     setImmediate      │
+       └───────────────────────┘
+
+note: each box will be referred to as a "phase" of the event loop.
+
+*There is a slight discrepancy between the Windows and the Unix/Linux
+implementation, but that's not important for this demonstration.  The most
+important parts are here.  There are actually seven or eight steps, but the
+ones we care about--ones that Node actually uses are these four.*
+
+## timers
+
+This phase executes callbacks scheduled by `setTimeout()` and `setInterval()`.
+When you create a timer, you make a call to setTimeout().  The event loop will
+eventually enter the 'poll' phase which determines how many milliseconds remain
+until the next timer. If there is a timer, it will wait for connections for that
+many milliseconds. After that many milliseconds, it will break the 'poll' phase
+and wrap back around to the timers phase where those callbacks can be processed.
```

Comment:
When the poll phase of the event loop is entered the number of ms before the soonest timer is to be called is set as the poll's timeout. Meaning the poll phase will return after "timeout" ms.

Take note that the poll phase can only return while idle. Meaning execution of a callback is allowed to run to completion, and can cause unexpected delay running the timer.